analysis/src/Files.ml

```diff
@@ -84,16 +84,26 @@ let rec collectDirs path =
        |> List.concat)
   | _ -> []
 
-let rec collect ?(checkDir = fun _ -> true) path test =
-  match maybeStat path with
-  | None -> []
-  | Some {Unix.st_kind = Unix.S_DIR} ->
+let rec collect ?(checkDir = fun _ -> true) ?maxDepth path test =
+  match (maxDepth, maybeStat path) with
+  | None, None -> []
+  | Some 0, _ -> []
+  | None, Some {Unix.st_kind = Unix.S_DIR} ->
     if checkDir path then
       readDirectory path
       |> List.map (fun name ->
              collect ~checkDir (Filename.concat path name) test)
       |> List.concat
     else []
+  | Some n, Some {Unix.st_kind = Unix.S_DIR} ->
+    if checkDir path then
+      readDirectory path
+      |> List.map (fun name ->
+             collect ~checkDir ~maxDepth:(n - 1)
```

Comment:
This needs to be made a bit more stable.

If users use a monorepo, they may still need to look upward to find the runtime package.